osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -257,9 +257,11 @@ public void EndPlacement(HitObject hitObject)
 
         public void Delete(HitObject hitObject) => EditorBeatmap.Remove(hitObject);
 
-        public override Vector2 GetSnappedPosition(Vector2 position) => distanceSnapGrid?.GetSnapPosition(position) ?? position;
+        public override Vector2 GetSnappedPosition(Vector2 position) => distanceSnapGrid?.GetSnappedPosition(position) ?? position;
 
-        public override double GetSnappedTime(double startTime, Vector2 position) => distanceSnapGrid?.GetSnapTime(position) ?? startTime;
+        public override double GetSnappedTime(double startTime, Vector2 position) => distanceSnapGrid?.GetSnappedTime(position) ?? startTime;
+
+        public override float GetSnappedDistance(float distance) => distanceSnapGrid?.GetSnappedDistance(distance) ?? distance;
```

Comment:
I'm a tad confused by this flow. This doesn't seem like the correct class to be using for this.

A visual grid is providing slider lengths makes me want to take a step back. I think I mentioned before, but I believe the beat snapping / velocity code should probably be in another place (maybe the `Beatmap` class or similar?), as it will be used in multiple cases (like the one in this PR).

Reply:
I think it makes perfect sense for a distance snapping grid to take a distance and snap that intervals on the grid. This is effectively doing the same thing, except that the physical display of the distance isn’t necessarily on the grid itself.

Can you explain where else distance snap will be used, if it were to be moved in the beatmap?

Also consider that calls to it likely won’t be able to go through the beatmap but the composer as it currently is, because a keypress or setting should globally disable the distance snap.